ersilia/core/tracking.py

```diff
@@ -318,10 +342,10 @@ def check_types(self, result_df, metadata):
         count = 0
 
         # ignore key and input columns
-        dtypes_list = result_df.loc[:, ~result_df.columns.isin(["key", "input"])].dtypes
+        dtypes_list = {col: result[col] for col in result if col not in ["key", "input"]}
 
-        for i in dtypes_list:
-            if type_dict[str(i)] != metadata["Output Type"][0]:
+        for column, dtype in dtypes_list.items():
+            if type_dict.get(str(dtype)) != metadata["Output Type"][0]:
```

Comment:
Isn't `dtype` a list here since `result[col]` is basically a list of all the values for a given column?